### ISSUE TITLE
(Voicing) Friction hints during interaction do not seem to fire correctly 

### ISSUE THREAD

**Issue by Matthew-Moore240** [phetsims/friction#194]
**During interaction**

When books are grabbed it should read 
_Move Chemistry book down to explore friction._

When books are grabbed Hint after all atoms (actually more than half) have been evaporated.
_Reset sim to make more observations._

**Hint Please button:**
Current interaction hint:
_Move Chemistry book down to explore friction._

Hint when the chemistry book is down but there are still lots of atoms left:
_Continue rubbing chemistry book to generate even more friction._

Hint after all atoms (actually more than half) have been evaporated:
_Reset sim to make more observations._

**Comment by terracoda** [comment]
@Matthew-Moore240, I am re-assigning this issue back to you. 
To document the interaction, you need to document the voicing separately for mouse/touch input and keyboard/gesture input.

We can discuss and then re-assign to @jessegreenberg.

**Comment by Matthew-Moore240** [comment]
This was in the design doc. Is this what we are thinking?

Mouse input (w/ interactive highlights on)

Mouse over pink highlight
Mouse Click:
Grabbed highlight (dotted pink highlight)
Object & text: Chemistry book grabbed. 
- Context:
- Help: 
- (if not touching physics book) Move down to rub harder.
- (if touch physics book) Rub fast or slow.

Touch input

On tap (dotted pink highlight appears):
Object & text: Chemistry book grabbed. 
- Context:
- Help: 
- (if not touching physics book) Move down to rub harder.
- (if touch physics book) Rub fast or slow.

Gesture input

Alternative Input (Keyboard) 
- Focus: Object name & current state
- Chemistry book. Lightly on Physics book.
- Look for grab buttons. Once grabbed, use letter keys W, A, S, or D to move book or zoomed-in book up, left, down, or right.

**Comment by terracoda** [comment]
@Matthew-Moore240, this is looking pretty good. Note that in voicing, the word "button" is never voiced, so the alternative input help, starting with "Look for grab buttons" will need some further thought. 

For this sim, I think about three categories of "hint content". Only 1 & 2 are specifically "hint content" in terms of teh voicing deign. There are three kinds of hints I designed fro non-visual access via the PDOM:
**1. Two general hints which are dynamic:** 
- Grab chemistry book to play.
- Reset sim to make more observations.

I see these kinds of hints as the hints that are good candidates for the Hint read-me button in the voicing design.


**2. Individual interaction help text hints** 
- The only example from the PDOM design for this sim is that long hint for the custom grab and drag interaction, 
    >  Look for grab buttons. Once grabbed, use letter keys W, A, S, or D to move book or zoomed-in book up, left, down, or right. 

This "direct interaction help text". In voicing design, direct hints on interactions are what gets turned via the checkbox in the Preferences menu.

This hint won't work for the voicing design because we don't use role names, like "button" in the voicing design. Sometimes we can use the help text straight from the PDOM design, but in this case we cannot. And often we want something shorter. That said, this interaction is a custom drag and grab, so I am not sure the best way to voice it yet. 

In the first interviews with blind learners it was referred to as a slider, even though it technically not a slider. I think that is good association for a learner to make - the book moves back and forth and they use the arrow keys to move it. It kind of works like a slider. The guidance needed for alternative input on the book before it is grabbed needs to be about how to grab. Ideally you need no instructions for that, but you might.

**3. Interaction guiding**
- These hints fire during interaction (actual moving the book) to get the learners rubbing the book efficiently and effectively. This sim was the first sim where I designed context responses to delivery direct interaction guidance (as opposed to delivery the sim's science information). For this sim, these hints are the strings that happen after a grab. Strings like "Rub fast or slow", "Move down to rub harder", and positional information like "left edge" and "right edge", and the reset hint is also fires once most or all of the atoms have vaporized. In voicing design, I think these kinds of hints will remain in the design regardless of whether or not the helpful hints checkbox is checked.


I think you need to think about if you need any specific help text about the grab interaction for alternative input & custom gesture. With alternative input on the first go around words actually show up on the screen, so you want to consider what is displayed, on the second go around the words may not be there, so you want to consider that, too.

**Comment by terracoda** [comment]
I am unassigning myself. @Matthew-Moore240, let me know if you need anything from me.

**Comment by Matthew-Moore240** [comment]
This seems fixed. Closing.
